tests/chgres_cube/ftst_program_setup_varmaps.F90

```diff
@@ -0,0 +1,59 @@
+! Unit test for chres_cube utility, reading varmap files. 
+!
+! Ed Hartnett 5/3/21
+
+program ftst_program_setup_varmaps
+  use mpi
+  use program_setup
+  implicit none
+  integer :: my_rank, nprocs
+  integer, parameter :: MAX_NAME_LEN = 20
+  integer, parameter :: EXPECTED_NUM_VARS = 23
+  integer, parameter :: EXPECTED_NUM_TRACERS = 7
+  character(len=MAX_NAME_LEN) :: expected_var_names(EXPECTED_NUM_VARS) = [character(len=20):: 'dzdt', 'sphum', 'liq_wat', &
+       'o3mr', 'ice_wat', 'rainwat', 'snowwat', 'graupel', 'vtype', 'sotype', 'vfrac', 'fricv', 'sfcr', 'tprcp', &
+       'ffmm', 'f10m', 'soilw', 'soill', 'soilt', 'cnwat', 'hice', 'weasd', 'snod']
+  character(len=MAX_NAME_LEN) :: expected_field_names(EXPECTED_NUM_VARS) = [character(len=20):: 'dzdt', 'sphum', &
+       'liq_wat', 'o3mr', 'ice_wat', 'rainwat', 'snowwat', 'graupel', 'vtype', 'stype', 'vfrac', 'uustar', 'zorl', &
+       'tprcp', 'ffmm', 'f10m', 'smc', 'slc', 'stc', 'cnwat', 'icetk', 'weasd', 'snod']
+  character(len=MAX_NAME_LEN) :: expected_missing_var_methods(EXPECTED_NUM_VARS) = [character(len=20):: 'set_to_fill', 'set_to_fill', &
+       'set_to_fill', 'set_to_fill', 'set_to_fill', 'set_to_fill', 'set_to_fill', 'set_to_fill', 'skip', 'skip', 'skip', 'set_to_fill', 'set_to_fill', &
+       'set_to_fill', 'set_to_fill', 'set_to_fill', 'stop', 'set_to_fill', 'stop', 'set_to_fill', 'set_to_fill', 'set_to_fill', 'set_to_fill']
+  real :: expected_missing_var_values(EXPECTED_NUM_VARS) = (/ 0.0, 1E-7, 0.0, 1E-7, 0.0, 0.0, 0.0, 0.0, 0.0, &
+       0.0, 0.0, 0.0, 0.01, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.5, 0.0, 0.0 /)
+  character(len=MAX_NAME_LEN) :: expected_tracers_input(EXPECTED_NUM_TRACERS) = [character(len=20):: 'sphum', 'liq_wat', &
+       'o3mr', 'ice_wat', 'rainwat', 'snowwat', 'graupel']
+  integer :: i
+  integer :: ierr
+
+  call mpi_init(ierr)
+  call MPI_Comm_rank(MPI_COMM_WORLD, my_rank, ierr)
+  call MPI_Comm_size(MPI_COMM_WORLD, nprocs, ierr)
+
+  if (my_rank .eq. 0) print*, "Starting test of program_setup reading varmaps."
+  if (my_rank .eq. 0) print*, "testing read_varmap with GFSphys_varmap.txt..."
+  varmap_file = "data/GFSphys_varmap.txt"
+  input_type = "grib2"
+  call read_varmap()
```

Comment:
Does your test check 'var_type'?

Reply:
var_type is a local variable that does not leave the subroutine. It's use determines the contents of the tracer array.

Reply:
@GeorgeGayno-NOAA do you know what var_types D and S are?

Reply:
D is what I used to denote variables processed by the atmosphere subroutine that aren't tracers, and S is for variables processed by surface